File: CodeKata/Looping - 14/Python/948. Add Integers without Carry.py
a, b = input().split()
a = a[::-1]
b = b[::-1]

max_len = max(len(a), len(b))
a = a.ljust(max_len, "0")
b = b.ljust(max_len, "0")

result = ""

for i in range(max_len):
    digit_sum = (int(a[i]) + int(b[i])) % 10
    result = str(digit_sum) + result

result = result.lstrip("0")

print(result if result else "0")

File: CodeKata/Looping - 14/JavaScript/948. Add Integers without Carry.js
const readline = require("readline");

const inp = readline.createInterface({
    input: process.stdin
});

const userInput = [];

inp.on("line", (data) => {
    userInput.push(data);
});

inp.on("close", () => {
    let [a, b] = userInput[0].split(" ");

    a = a.split("").reverse().join("");
    b = b.split("").reverse().join("");

    let max_len = Math.max(a.length, b.length);

    a = a.padEnd(max_len, "0");
    b = b.padEnd(max_len, "0");

    let result = "";

    for (let i = 0; i < max_len; i++) {
        let digitSum = (parseInt(a[i]) + parseInt(b[i])) % 10;
        result = digitSum.toString() + result;
    }

    result = result.replace(/^0+/, "");

    console.log(result === "" ? "0" : result);
});